### PR TITLE
feat(chat): add chat support to the client

### PR DESCRIPTION
**Pull request by pilartomas**

### CODE REVIEW

File: src/helpers/common.ts [hunk 1]
```diff
@@ -151,6 +151,25 @@ export function callbackifyGenerator<T>(generatorFn: () => AsyncGenerator<T>) {
...
 }
 
+export function callbackifyStream<T>(
+  stream: Readable,
+  callbackFn: Callback<T>,
+) {
+  stream.on('data', (data) => callbackFn(null, data));
+  stream.on('error', (err) => (callbackFn as ErrorCallback)(err));
+  stream.on('finish', () => (callbackFn as DataCallback<T | null>)(null, null));
+}
```
Comment (Tomas2D) [inline src/helpers/common.ts:161]: Based on naming, it should be a factory function. (look at built-in NodeJS util `.callbackify` -> returns a function`

```typescript
export function callbackifyStream<T>(callbackFn: Callback<T>) {
  return (stream: Readable) => {
    stream.on('data', (data) => callbackFn(null, data));
    stream.on('error', (err) => (callbackFn as ErrorCallback)(err));
    stream.on('finish', () => (callbackFn as DataCallback<T | null>)(null, null));
  }
}
```

File: src/helpers/common.ts [hunk 2]
```diff
@@ -151,6 +151,25 @@ export function callbackifyGenerator<T>(generatorFn: () => AsyncGenerator<T>) {
...
+
+export function callbackifyPromise<T>(
```
Comment (Tomas2D) [inline src/helpers/common.ts:163]: same here

File: src/tests/e2e/client.test.ts [hunk 3]
```diff
@@ -180,6 +184,105 @@ describe('client', () => {
...
+          stream.on('data', (chunk) => {
+            validateStreamChunk(chunk);
+            chunks.push(chunk);
+          });
+          stream.on('close', () => {
+            resolve(chunks);
+          });
+          stream.on('error', (err) => {
+            reject(err);
+          });
```
Comment (Tomas2D) [inline src/tests/e2e/client.test.ts:220]: We could maybe use the `for await loop` syntax for testing.

**Review by Tomas2D — CHANGES_REQUESTED** [review]

File: src/helpers/common.ts (hunk 1 above)
Comment (pilartomas) [reply src/helpers/common.ts:161]: Will do 👍 , note your example here has the arguments reversed.

**Review by Tomas2D — APPROVED** [review]
👍🏻

**Pull request merged**